apps/extension/src/background/keyring/types.ts

```diff
@@ -0,0 +1,32 @@
+export type Bip44Path = {
+  account: number;
+  change: number;
+  index: number;
+};
+
+export type MnemonicState = {
+  id: string;
+  alias?: string;
+  phrase: Uint8Array;
+};
+
+export type DerivedAccount = {
+  id: string;
+  address: string;
+  alias?: string;
+  bip44Path: Bip44Path;
+  establishedAddress?: string;
+};
+
+export type AccountState = DerivedAccount & {
+  parentId: string;
+  private: Uint8Array;
+  public: Uint8Array;
+};
+
+export enum KeyRingStatus {
+  NOTLOADED,
+  EMPTY,
+  LOCKED,
+  UNLOCKED,
```

Comment:
nit: all caps feels inconsistent against the other other enums
```suggestion
  NotLoaded,
  Empty,
  Locked,
  Unlocked,
```